File: backend/routes/order.py
import base64
from flask import Blueprint, request, make_response
from datetime import datetime

from common.mysql_connector import db
from common.auth import auth

order = Blueprint("order", __name__)

@order.route('/startorder', methods=["POST"])
@auth
def start_order():
    data = request.form.to_dict()
    user = data.get("user")
    name = data.get("name")
    description = data.get("description")
    contact = data.get("contact")
    price = data.get("price")
    img_file = data.get("imgFile")

    if name is None or description is None or contact is None or price is None or price is None or img_file is None:
        return make_response("Information not complete", 403)

    starttime = datetime.now().strftime("%Y%m%d%H%M%S")
    cursor = db.cursor()
    try:
        sql = f"INSERT INTO Item(name, description, contact, price, seller, selldate) VALUES ('{name}', '{description}', '{contact}', '{price}', '{user}', {starttime});"
        cursor.execute(sql)
        cursor.execute(f"SELECT LAST_INSERT_ID();")
        item_id = cursor.fetchone()[0]
        sql = f"INSERT INTO `Relation`(`username`, `item_id`) VALUES ('{user}', {item_id});"
        cursor.execute(sql)

        split_img_file = img_file.split(';base64,')
        main_data = split_img_file[1]
        suffix = split_img_file[0].split('/')[-1]
        img_path = f"img/{item_id}.{suffix}"
        img_data = base64.b64decode(main_data)
        file = open(img_path, 'wb')
        file.write(img_data)

        sql = f"UPDATE Item SET img_path='{img_path}' WHERE id={item_id};"
        cursor.execute(sql)

        db.commit()
        response = make_response("success", 200)
    except Exception as e:
        db.rollback()
        response = make_response("Internal Error", 500)
        print(e)
    cursor.close()
    return response

File: frontend/src/api/order.js
const SERVER_ADDRESS = 'http://0.0.0.0:5002'

import axios from 'axios'

export default {
  start (token, user, name, description, contact, price, imgFile, callback) {
    let form = new FormData()
    form.append('token', token)
    form.append('user', user)
    form.append('name', name)
    form.append('description', description)
    form.append('contact', contact)
    form.append('price', price)
    form.append('imgFile', imgFile)
    axios.post(SERVER_ADDRESS + '/startorder', form)
      .then(response => callback(response.status))
      .catch(error => callback(error.response.status))
  },
  get (token, user, cbSuccess, cbFail) {
    let form = new FormData()
    form.append('token', token)
    form.append('user', user)
    axios.post(SERVER_ADDRESS + '/getorder', form)
      .then(response => cbSuccess(response.data))
      .catch(error => cbFail(error.response.status))
  },
  getSpecific (token, user, order_id, cbSuccess, cbFail) {
    let form = new FormData()
    form.append('token', token)
    form.append('user', user)
    form.append('order_id', order_id)
    axios.post(SERVER_ADDRESS + '/getspecificorder', form)
      .then(response => cbSuccess(response.data))
      .catch(error => cbFail(error.response.status))
  }
}
